### ISSUE TITLE
AttrDict is old and appears to be broken on python3.10

### ISSUE THREAD

**Issue by aoberai** [kc-ml2/SimpleDreamer#12]
Mentions "cannot import name 'Mapping' from 'collections'" is an issue due to python version: https://stackoverflow.com/a/70557518. This issue comes from the python dependency attrdict, which is now old and public archived: https://github.com/bcj/AttrDict. Due to a lack in update from attrdict to resolve this issue, as such, the repo no longer works when running the default repo example. Didn't investigate too deeply so might be an issue on my side.

**Comment by aoberai** [comment]
As a side note, appreciate you writing this @seolhokim. Found it helps to read the repo with the original dreamerv1 paper and subsequent versions.

**Comment by aoberai** [comment]
Made a quick little monkey patch fix here to attrdict so you can still use it without finding a replacement just to keep this repo working (since there doesn't seem to be any future long term dev planned on it). It now works for me.

https://github.com/kc-ml2/SimpleDreamer/pull/13

**Comment by aoberai** [comment]
Aside from this, don't want to create a whole pr for this if not requested but I got this error from running the repo: https://stackoverflow.com/questions/72441758/typeerror-descriptors-cannot-not-be-created-directly

This solution worked well. It's a quick fix. You can add it to requirements.txt or I can do it as well if wanted.

pip install 'protobuf<=3.20.1' --force-reinstall

**Comment by seolhokim** [comment]
I tried to test it in a Python 3.10 environment, but I can't even install gym==0.19. So, I think we need to fix lots of things to upgrade the version.

**Comment by aoberai** [comment]
Gymnasium is the new replacement, those are relatively easy fixes. I got it working, can upload a new requirements.txt if you would like.

**Comment by seolhokim** [comment]
I would appreciate it if you could do that.

**Comment by aoberai** [comment]
Sounds good, I can create a separate PR for that once #13 is merged.

**Comment by seolhokim** [comment]
#13 merged.

**Comment by aoberai** [comment]
Sg, made #14 as per your comment and #15 (requirements.txt although I recommend verifying it yourself).

**Comment by seolhokim** [comment]
Thank you so much for your valuable contribution! With your help, we can now effectively maintain it🥇
